src/modules/dashboard/components/Whitelist/WhitelistAddresses/UploadAddressesExtensionWidget.tsx

```diff
@@ -0,0 +1,124 @@
+
+const MSG = defineMessages({
+  inputLabel: {
+    id: `dashboard.Whitelist.WhitelistAddresses.UploadAddressesExtensionWidget.inputLabel`,
+    defaultMessage: 'Add wallet address that has passed KYC/AML',
+  },
+  agreement: {
+    id: `dashboard.Whitelist.WhitelistAddresses.UploadAddressesExtensionWidget.agreement`,
+    defaultMessage: 'Agreement',
+  },
+});
+
+interface Props {
+  colony: Colony;
+  userHasPermission: boolean;
+  whitelistAgreementHash?: string | null;
+}
+
+const UploadAddressesExtensionWidget = ({
+  colony: { colonyAddress },
+  userHasPermission,
+  whitelistAgreementHash,
+}: Props) => {
+  const [showInput, setShowInput] = useState<boolean>(true);
+  const toggleShowInput = () => setShowInput(!showInput);
+  const history = useHistory();
+  const openAgreementDialog = useDialog(AgreementDialog);
+
+  const openDialog = useCallback(
+    () =>
+      whitelistAgreementHash &&
+      openAgreementDialog({
+        agreementHash: whitelistAgreementHash,
+      }),
+    [openAgreementDialog, whitelistAgreementHash],
+  );
+
+  const transform = useCallback(
+    pipe(
+      mapPayload(({ whitelistAddress, whitelistCSVUploader }) => {
+        return {
+          userAddresses:
+            whitelistAddress !== undefined
+              ? [whitelistAddress]
+              : whitelistCSVUploader[0].parsedData,
+          colonyAddress,
+          status: true,
+        };
+      }),
+      withMeta({ history }),
+    ),
+    [],
+  );
+
+  return (
+    <ActionForm
+      initialValues={{}}
+      validationSchema={
+        showInput ? validationSchemaInput : validationSchemaFile
+      }
+      submit={ActionTypes.WHITELIST_UPDATE}
+      error={ActionTypes.WHITELIST_UPDATE_ERROR}
+      success={ActionTypes.WHITELIST_UPDATE_SUCCESS}
+      transform={transform}
+    >
+      {({ errors, isValid, isSubmitting }: FormikProps<{}>) => (
+        <>
+          <div className={styles.container}>
+            <UploadAddresses
+              userHasPermission={userHasPermission}
+              errors={errors}
+              isSubmitting={isSubmitting}
+              showInput={showInput}
+              toggleShowInput={toggleShowInput}
+            />
+          </div>
+          <div className={styles.buttonsContainer}>
+            <div className={styles.agreeemntButton}>
+              {whitelistAgreementHash && (
+                <Button
+                  appearance={{ theme: 'blue' }}
+                  onClick={openDialog}
+                  text={MSG.agreement}
+                />
+              )}
+            </div>
+            <Button
+              appearance={{ theme: 'primary', size: 'large' }}
+              text={{ id: 'button.confirm' }}
+              disabled={!userHasPermission || !isValid || isSubmitting}
+              type="submit"
+              loading={isSubmitting}
+            />
+          </div>
+        </>
+      )}
+    </ActionForm>
+  );
+};
```

Comment:
If we will be using this component in both cases, its need to be adjusted to fit this new "whitelist concept"

Reply:
We will be using `UploadAddresses` in both situations. 
Perhaps I am not aware of all the details of the new "whitelist concept", but `UploadAddresses` should support both Whitelisting. If there is some functionality missing, or if you foresee any issues, please let me know.